MoppApp/MoppApp/ContainerActions.swift

```diff
@@ -162,42 +201,64 @@ extension ContainerActions where Self: UIViewController {
+        if (landingViewController.containerType == .asic) {
+            MoppLibContainerActions.sharedInstance().createContainer(
+                withPath: containerPath,
+                withDataFilePaths: dataFilePaths,
+                success: { container in
+                    if cleanUpDataFilesInDocumentsFolder {
+                        cleanUpDataFilesInDocumentsFolderCode()
+                    }
+                    if container == nil {
+                        
+                        landingViewController.importProgressViewController.dismissRecursively(animated: false, completion: nil)
+                        
+                        let alert = UIAlertController(title: L(.fileImportCreateNewFailedAlertTitle), message: L(.fileImportCreateNewFailedAlertMessage, [fileName]), preferredStyle: .alert)
                         alert.addAction(UIAlertAction(title: L(.actionOk), style: .default, handler: nil))
-
-                    landingViewController.present(alert, animated: true)
-                    return
-                }
-                
-                let containerViewController = ContainerViewController.instantiate()
+                        
+                        landingViewController.present(alert, animated: true)
+                        return
+                    }
+                    
+                    let containerViewController = SigningContainerViewController.instantiate()
                     containerViewController.containerPath = containerPath
+                    
                     containerViewController.isCreated = true
                     containerViewController.startSigningWhenOpened = startSigningWhenCreated
-                
-                landingViewController.importProgressViewController.dismissRecursively(animated: false, completion: {
-                    navController?.pushViewController(containerViewController, animated: true)
-                })
-            
+                    
+                    landingViewController.importProgressViewController.dismissRecursively(animated: false, completion: {
+                        navController?.pushViewController(containerViewController, animated: true)
+                    })
+                    
             }, failure: { error in
                 if cleanUpDataFilesInDocumentsFolder {
                     cleanUpDataFilesInDocumentsFolderCode()
                 }
                 landingViewController.importProgressViewController.dismiss(animated: false, completion: nil)
                 MoppFileManager.shared.removeFile(withPath: filePath)
             }
-        )
+            )
+        }else{
+
+            let containerViewController = CryptoContainerViewController.instantiate()
+            let container = CryptoContainer.init(filename: containerFilename as NSString, filePath: containerPath as NSString)
+            containerViewController.containerPath = containerPath
+            
+            dataFilePaths.forEach {
```

Comment:
I would prefer using unnamed closure parameter specifier (`$n`) only if there is a very small closure implementation.